### PR TITLE
raft: broadcast_tables: add support for bind variables

### PR DESCRIPTION
**Pull request by margdoc**
Extended the queries language to support bind variables which are bound in the execution stage, before creating a raft command.
    
Adjusted `test_broadcast_tables.py` to prepare statements at the beginning of the test.

Fixed a small bug in `strongly_consistent_modification_statement::check_access`.

### CODE REVIEW

**Comment by margdoc** [comment]
Based on https://github.com/scylladb/scylladb/pull/11164

**Comment by margdoc** [comment]
@kbr- please review

File: cql3/operation.hh [hunk 1]
```diff
@@ -19,12 +20,12 @@
...
 #include <optional>
 
-namespace service::broadcast_tables {
-    class update_query;
-}
-
 namespace cql3 {
 
+namespace statements {
+    struct prepared_update_query;
```
Comment (kbr-) [inline cql3/operation.hh:26]: I would suggest a more specific name to make it clear that this is specific to broadcast tables / strongly consistent statements. Or namespacing.
e.g. `cql3::statements::broadcast_tables::prepared_update` or `cql3::statements::strongly_consistent::prepared_update` or something like that.

File: cql3/statements/strongly_consistent_modification_statement.cc [hunk 2]
```diff
@@ -91,7 +111,7 @@ uint32_t strongly_consistent_modification_statement::get_bound_terms() const {
 future<> strongly_consistent_modification_statement::check_access(query_processor& qp, const service::client_state& state) const {
     const data_dictionary::database db = qp.db();
     auto f = state.has_column_family_access(db, _schema->ks_name(), _schema->cf_name(), auth::permission::MODIFY);
-    if (_query.value_condition->has_value()) {
+    if (_query.value_condition.has_value()) {
```
Comment (kbr-) [inline cql3/statements/strongly_consistent_modification_statement.cc:114]: So this was a bug?

File: cql3/statements/strongly_consistent_select_statement.hh [hunk 3]
```diff
@@ -10,17 +10,22 @@
...
 
+#include "cql3/expr/expression.hh"
 #include "cql3/statements/select_statement.hh"
 #include "service/broadcast_tables/experimental/lang.hh"
 
 namespace cql3 {
 
 namespace statements {
 
+struct prepared_select_query {
```
Comment (kbr-) [inline cql3/statements/strongly_consistent_select_statement.hh:21]: Same naming/namespacing issue as for updates.

**Review by kbr- — CHANGES_REQUESTED** [review]

File: cql3/statements/strongly_consistent_select_statement.hh (hunk 3 above)
Comment (margdoc) [reply cql3/statements/strongly_consistent_select_statement.hh]: Done.

File: cql3/operation.hh (hunk 1 above)
Comment (margdoc) [reply cql3/operation.hh]: Renamed to `cql3::statements::broadcast_tables::prepared_update`.

File: cql3/statements/strongly_consistent_modification_statement.cc (hunk 2 above)
Comment (margdoc) [reply cql3/statements/strongly_consistent_modification_statement.cc:114]: Apparently yes.

**Review by kbr- — APPROVED** [review]

**Pull request closed without merging**